### ISSUE TITLE
Bug: `isPathIgnored` returns ignored files

### ISSUE THREAD

**Issue by fudom** [eslint/eslint#18556]
### Environment

Node version: 20.14.0
npm version: 10.7.0
Local ESLint version: 9.4.0 
Global ESLint version: -
Operating System: Win 10

### What parser are you using?

Default (Espree)

### What did you do?

```js
const removeIgnoredFiles = async (files) => {
  const eslint = new ESLint();
  const isIgnored = await Promise.all(
    files.map((file) => {
      return eslint.isPathIgnored(file);
    })
  );
  const filteredFiles = files.filter((_, i) => !isIgnored[i]);
  return filteredFiles.join(' ');
};
```

**eslint.config.mjs**

```js
{
    ignores: ['./*', '!./src', '**/*.spec.ts', './src/assets/*'],
}
```
The `ignores` property without other property is global. ([Docs](https://eslint.org/docs/latest/use/configure/configuration-files#globally-ignoring-files-with-ignores))

The CLI works. But ESLint isPathIgnored still returns e.g. `./jest.config.js` from root.

It's for git hook husky lintstage because eslint otherwise ignores it too.

**.lintstagedrc.js**
```js
module.exports = {
  '*.{ts,js}': async (files) => {
    const filesToLint = await removeIgnoredFiles(files);
    return [`eslint --fix --max-warnings=0 ${filesToLint}`];
  },
};
```

### What did you expect to happen?

Don't process ignored files.

### What actually happened?

It still returns ignored files. Btw. the eslint extension of vscode does also report it (ignores `ignores`).

### Link to Minimal Reproducible Example

Not needed in this case.

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Please provide a minimal reproducible example, as we can't investigate any bug reports without a repro case.

> ```js
> {
>     ignores: ['./*', '!./src', '**/*.spec.ts', './src/assets/*'],
> }
> ```

Note that patterns are relative to the base path (location of your eslint config file) and should not start with `./`. If you remove `./` from the patterns, you'll likely get the expected behavior.

```js
{
    ignores: ['*', '!src', '**/*.spec.ts', 'src/assets/*'],
}
```

> **.lintstagedrc.js**
> 
> ```js
> module.exports = {
>   '*.{ts,js}': async (files) => {
>     const filesToLint = await removeIgnoredFiles(files);
>     return [`eslint --fix --max-warnings=0 ${filesToLint}`];
>   },
> };
> ```

For this use case, you could use [`--no-warn-ignored`](https://eslint.org/docs/latest/use/command-line-interface#--no-warn-ignored) CLI option instead of manually filtering out ignored files.
